pkg/storage/storage.go

```diff
@@ -2,122 +2,96 @@ package storage
+	return &storage{kubeClient: kubeClient}
+}
 
-	cm, err := GetConfigMap(ins.Namespace, ins.Name)
+func (s *storage) CheckConfigMapEntry(key string, ins types.NamespacedName) (string, error) {
+	cm, err := s.getConfigMap(ins.Namespace, ins.Name)
 	if err != nil {
 		return "", err
 	}
 
-	data, found, err := unstructured.NestedMap(cm.Object, "data")
-	if err != nil {
-		return "", fmt.Errorf("error getting the Data field: %v", err)
-	}
-
-	if !found {
-		return "", nil
-	}
-
-	if value, found := data[key]; found {
-		return value.(string), nil
-	}
-
-	return "", nil
+	return cm.Data[key], nil
 }
 
-func UpdateConfigMapEntry(key string, value string, ins types.NamespacedName) error {
-
-	cm, err := GetConfigMap(ins.Namespace, ins.Name)
+func (s *storage) UpdateConfigMapEntry(key string, value string, ins types.NamespacedName) error {
+	cm, err := s.getConfigMap(ins.Namespace, ins.Name)
 	if err != nil {
 		warn.OnError(err)
 		return err
 	}
 
-	// Just looking if exists so we can create or update
-	entries, found, err := unstructured.NestedMap(cm.Object, "data")
-	if err != nil {
-		return err
-	}
-
-	if !found {
-		entries = make(map[string]interface{})
+	if cm.Data == nil {
+		cm.Data = make(map[string]string)
 	}
 
-	entries[key] = value
+	if cm.Data[key] != value {
+		cm.Data[key] = value
 
-	if err = unstructured.SetNestedMap(cm.Object, entries, "data"); err != nil {
-		return err
+		if err = s.updateObject(context.TODO(), cm); err != nil {
+			warn.OnError(err)
+			return err
+		}
 	}
 
-	err = clients.Interface.Update(context.TODO(), cm)
-	if err != nil {
-		warn.OnError(err)
-		return err
-	}
 	return nil
 }
 
-func DeleteConfigMapEntry(delete string, ins types.NamespacedName) error {
-
-	cm, err := GetConfigMap(ins.Namespace, ins.Name)
+func (s *storage) DeleteConfigMapEntry(key string, ins types.NamespacedName) error {
+	cm, err := s.getConfigMap(ins.Namespace, ins.Name)
 	if err != nil {
 		warn.OnError(err)
 		return err
 	}
 
-	// Just looking if exists so we can create or update
-	old, found, err := unstructured.NestedMap(cm.Object, "data")
-	if err != nil {
-		return err
-	}
+	if _, ok := cm.Data[key]; ok {
+		delete(cm.Data, key)
 
-	if !found {
-		return nil
+		if err = s.updateObject(context.TODO(), cm); err != nil {
+			warn.OnError(err)
+			return err
+		}
 	}
 
-	newMap := make(map[string]interface{})
+	return nil
+}
 
-	for k, v := range old {
-		if delete != k {
-			newMap[k] = v
-		}
-	}
+func (s *storage) getConfigMap(namespace string, name string) (*v1.ConfigMap, error) {
+	cm := &v1.ConfigMap{}
+	dep := types.NamespacedName{Namespace: namespace, Name: name}
 
-	if err = unstructured.SetNestedMap(cm.Object, newMap, "data"); err != nil {
-		return err
-	}
+	err := s.kubeClient.Get(context.TODO(), dep, cm)
 
-	err = clients.Interface.Update(context.TODO(), cm)
-	if err != nil {
+	if apierrors.IsNotFound(err) {
 		warn.OnError(err)
-		return err
+		return cm, err
```

Comment:
Any reason `cm` is returned instead of `nil`?

Reply:
Nice catch! While calling functions should anyway not use the return value, I changed it to `nil`.